pkg/scheduler/framework/plugins/podtopologyspread/scoring.go

```diff
@@ -147,7 +148,8 @@ func (pl *PodTopologySpread) PreScore(
 		// (1) `node` should satisfy incoming pod's NodeSelector/NodeAffinity
 		// (2) All topologyKeys need to be present in `node`
 		match, _ := requiredNodeAffinity.Match(node)
-		if !match || !nodeLabelsMatchSpreadConstraints(node.Labels, state.Constraints) {
+		if !isSystemDefaultConstraints(pod, pl.defaultConstraintsType) &&
```

Comment:
Ditto. The result of isSystemDefaultConstraints() is deterministic regardless of the internal loop.